components/MainColor/index.js

```diff
@@ -1,8 +1,29 @@
-export default function MainColor({ text, id }) {
+import { Fragment } from "react";
+import { StyledInput, StyledDiv, StyledLabel } from "./styles";
+
+export default function MainColor({ color, handleColorChange }) {
+  const colors = [
+    { id: "color1", text: "Color1", name: "color1" },
+    { id: "color2", text: "Color2", name: "color2" },
+    { id: "color3", text: "Color3", name: "color3" },
+  ];
+
   return (
-    <>
-      <label htmlFor={id}>{text}</label>
-      <input type="color" id={id} />
-    </>
+    <StyledDiv>
+      {colors.map(({ id, text, name }) => (
+        <Fragment key={id}>
+          <StyledLabel htmlFor={id}>
+            {text}
+            <StyledInput
+              type="color"
+              id={id}
+              name={name}
+              value={color[id]}
+              onChange={(event) => handleColorChange(id, event.target.value)}
+            />
+          </StyledLabel>
+        </Fragment>
+      ))}
+    </StyledDiv>
```

Comment:
Since you're only rendering a single child within the `Fragment`( ➡️ `StyledLabel`), there's no need to use it. 
The key prop can be applied directly to the StyledLabel component.

```suggestion
       <StyledDiv>
      {colors.map(({ id, text, name }) => (
        <StyledLabel key={id} htmlFor={id}>
          {text}
          <StyledInput
            type="color"
            id={id}
            name={name}
            value={color[id]}
            onChange={(event) => handleColorChange(id, event.target.value)}
          />
        </StyledLabel>
      ))}
    </StyledDiv>
```